### PR TITLE
Remove email from logging

### PR DESCRIPTION
**Pull request by jdfox**
PR
# Motivation and Context

From Technical Debt ticket "Remove email and name logging from party (2)", change was needed to remove excessive user detail from logs

# What has changed

In util.py, added handling of short, malformed and missing addresses.

No refactoring was required.

Unit test added to test_util.py 
# How to test?

make test
# Links

https://trello.com/c/vWfVa8OK/1463-remove-email-and-name-logging-from-party-2

### CODE REVIEW

File: test/test_util.py [hunk 1]
```diff
@@ -18,3 +18,23 @@ def test_flatten_keys_produces_expected_results(self):
...
+        testAddresses = {
+            "example@example.com": "e*****e@e*********m",
+            "prefix@domain.co.uk": "p****x@d**********k",
+            "first.name@place.gov.uk": "f********e@p**********k",
+            "me+addition@gmail.com": "m*********n@g*******m",
+            "a.b.c.someone@example.com": "a***********e@e*********m",
+            "john.smith123456@londinium.ac.co.uk": "j**************6@l****************k",
+            "me!?@example.com": "m**?@e*********m",
+            "m@m.com": "m@m***m",
+            "joe.bloggs": "j********s",
```
Comment (insacuri) [inline test/test_util.py:34]: Are these last 3 cases why you changed the function?  If so, it's not really necessary as valid emails are <thing>@<domain>.<tld>
If that is the case, could you change the function back to the simpler version?

File: ras_party/support/util.py [hunk 2]
```diff
@@ -64,7 +68,20 @@ def obfuscate_email(email):
...
+            prefix = f'{m[0][0]}{"*"*(len(m[0])-2)}{m[0][-1]}'
+        else:
+            prefix = m[0]
+        # If the domain is missing or 1 character, then we can't obfuscate it
+        if len(m) > 1 and len(m[1]) > 1:
+            domain = f'{m[1][0]}{"*"*(len(m[1])-2)}{m[1][-1]}'
+            return f'{prefix}@{domain}'
+        else:
+            return f'{prefix}'
+    except AssertionError:
```
Comment (insacuri) [inline ras_party/support/util.py]: How does the assertionError get raised?

**Review by insacuri — CHANGES_REQUESTED** [review]

File: test/test_util.py (hunk 1 above)
Comment (jdfox) [reply test/test_util.py:34]: I added that last one ("joe.bloggs") to test_util.py, for completeness, but when we spun up dev it turned out that malformed addresses (no @domain etc) are rejected by the UI, so probably not needed. Happy to trim that test list. 
But most of the extended checking was added in order to allow obfuscation on the existing tests for malformed addresses in ras-party/test/**test_respondent_controller.py** to pass in a "make test", e.g. lines 206, 211, etc. Want me to chop those too, or is that out of scope here?

File: ras_party/support/util.py (hunk 2 above)
Comment (jdfox) [reply ras_party/support/util.py]: That was also raised by test_respondent_controller.py (see above), so that may also be overkill.

Comment (jdfox) [reply ras_party/support/util.py]: I've replaced the try-except with a simpler set of ifs.

**Review by insacuri — APPROVED** [review]

**Review by warrenbailey — APPROVED** [review]

**Pull request merged**